src/BlazorWebView/src/Maui/Android/WebKitWebViewClient.cs

```diff
@@ -35,13 +37,33 @@ public override bool ShouldOverrideUrlLoading(AWebView? view, IWebResourceReques
 		{
 			// Handle redirects to the app custom scheme by reloading the URL in the view.
 			// Handle navigation to external URLs using the system browser, unless overriden.
-			var requestUri = request?.Url?.ToString();
+			if (request?.Url is { } url)
+			{
+				if (url.Scheme is "http" or "https")
+				{
+					if (TryOverrideHttpUrlLoading(view, request, url.ToString()))
+					{
+						return true;
+					}
+				}
+				else
+				{
+					TryStartActivityFromUri(url.ToString(), uriHasIntentScheme: url.Scheme == "intent");
+					return true;
```

Comment:
It's a bit subtle that `TryStartActivityFromUri` has that name (`Try...`) yet returns `void`, and that on this branch of the if/else we always return yet on the other we fall through to the base. I'm not suggesting the logic is wrong, just that a year from now, it might be a challenge for us to reason about why it goes like this. The fallthrough structure heightens the risk that we make a change that intends to affect one case but actually affects other cases too.

Would it be viable to do either one or both of the following?

 * Structure the code so there's no fallthrough from one case to another, i.e., that we're in "HTTP URL" or "other scheme URL" or "null URL" mode, and each of those fully encapsulates its own logic and definitely returns (which might involve returning a call to base.ShouldOverrideUrlLoading)
 * Add code comments briefly stating why the choice of whether or not to call base is correct. Perhaps each of the three top-level branches could have a sentence summarizing the reasoning for that case.

Likewise I'm wondering if we should be calling the `ExternalNavigationStarting` event for non-HTTP(S) destinations. My sense is that developers would likely want to have control over such navigations too.